packages/paypal-commerce-integration/src/PaypalCommerceRatePayPaymentMethod.tsx

```diff
@@ -79,6 +80,7 @@ const PaypalCommerceRatePayPaymentMethod: FunctionComponent<any> = ({
                 paypalcommerceratepay: {
                     container: '#checkout-payment-continue',
                     legalTextContainer: 'legal-text-container',
+                    loadingSpinner: ()  => LoadingSpinnerAnimation({}),
```

Comment:
I don't really sure that we should pass the whole component to the checkout sdk strategy and combine UI component logic between 2 parts (overlay on `checkout-sdk` side, loading component on `checkout-js` side)... 

`checkout-js` should be responsible for UI part of the implementation (to decide what should be done on UI side when the process is in pending state),  while `checkout-sdk` should tell `checkout-js` when to trigger that implementation.

So in this case my proposal is:
- to pass onPending callback to checkout-sdk and trigger it there whenever needed;
- implement the loader logic with overlay and spinner on `checkout-js` side.

It will help to keep our code simple to understand. Also we can describe in checkout-sdk what `onPending(isPending: true|false)` callback used for.

Whad do you think? @andriiVitvitskyi1990 @animesh1987 @bc-peng

Reply:
Yes. So when you call onPending callback in checkout sdk strategy, then checkout-js should show the loader to the customer. It will give other developers some flexibility for cases if they want to log something or do something else, when transaction is in pending status.

We should move loading UI responsibilities to checkout-js